src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -43,40 +47,47 @@ export const PasswordInput: React.FC<PasswordInputProps> = (props) => {
             <div className={b('additional-right-content')}>
                 {rightContent}
                 {value && showCopyButton ? (
-                    <CopyToClipboard text={String(value)} timeout={500}>
-                        {(state) => (
-                            <Button view="flat-secondary" className={b('button')} size="s">
-                                <Icon
-                                    size={14}
-                                    data={
-                                        state === CopyToClipboardStatus.Pending ? Copy : CopyCheck
-                                    }
-                                />
-                            </Button>
-                        )}
-                    </CopyToClipboard>
+                    <ClipboardButton
+                        text={value}
+                        hasTooltip={hasTooltip}
+                        size={16}
+                        className={b('copy-button')}
+                    />
                 ) : null}
-                {showVisibilityButton ? (
-                    <Button
-                        view="flat-secondary"
-                        onClick={onClick}
-                        className={b('button')}
-                        size="s"
+                {showRevealButton ? (
+                    <Tooltip
+                        disabled={!hasTooltip}
+                        content={
+                            hideValue ? i18n('label_show-password') : i18n('label_hide-password')
+                        }
                     >
-                        <Icon data={hideValue ? Eye : EyeSlash} size={14} />
-                    </Button>
+                        <Button
+                            view="flat-secondary"
+                            onClick={onClick}
+                            size={size}
```

Comment:
TextInput size !== Button size, take a look at clear button, maybe reuse this [logic](https://github.com/gravity-ui/uikit/blob/e9aff632b924307e7700e76855ba15abc905be3b/src/components/controls/common/ClearButton/ClearButton.tsx#L23) here as well